### ISSUE TITLE
Accuracy always 1 when using LSTM

### ISSUE THREAD

**Issue by ankeshanand** [fchollet/keras#2172]
Hi, 

I am modelling a simple text classification task with 3 target labels. I am using word2vec embeddings as input to the LSTM layer. 
`sequence_size` is the size of each sequence. `dimsize` is the word2vec dimension. 
`X_train` is a 3D tensor of the shape `(n_samples, sequence_size, dimsize)`

```
model = Sequential()
model.add(LSTM(400, input_shape=(sequence_size, dimsize)))
model.add(Dropout(0.5))
model.add(Dense(1))
model.add(Activation('sigmoid'))
model.compile(loss='binary_crossentropy',
              optimizer='adam')

batch_size = 32
model.fit(X_train, y_train, batch_size=batch_size, nb_epoch=3,
          validation_data=(X_test, y_test), show_accuracy=True)
score, acc = model.evaluate(X_test, y_test,
                            batch_size=batch_size,
                            show_accuracy=True)
```

But while training this model, I always get accuracy scores of 1 like this:

`
Epoch 1/3
31168/47226 [==================>...........] - ETA: 2142s - loss: -8.0915 - acc: 1.0000
`

What could be the issue here?
PS: I am using Keras 0.3.2

**Comment by NasenSpray** [comment]
> I am modelling a simple text classification task with 3 target labels.

The model in your has only one output...

> `y_train` is a simple vector of labels like [0,2,1,0,1,0,...]

`y_train` has to be >= 2D, e.g. `(n_samples, output_dimsize)`

**Comment by ankeshanand** [comment]
Thanks @NasenSpray, that solves it. I didn't find it mentioned somewhere in the docs though, and having used sklearn for long, I just assumed `y` to be one dimensional.

**Comment by yw411** [comment]
@ankeshanand，I am doing Chinese text setiment classification（positive and negative）and plan to use word2vec、LSTM 。I want to know what your accuracy  can reach，more than 90%？

**Comment by snowxiaoru** [comment]
I wan to know how do you put your wordvec into embedding level

**Comment by liormagen** [comment]
Can you share your entire code plz? the fixed one
